packages/@aws-cdk-testing/framework-integ/test/aws-elasticloadbalancingv2-actions/test/integ.cognito.ts

```diff
@@ -74,17 +158,61 @@ class CognitoStack extends Stack {
       conditions: [elbv2.ListenerCondition.pathPatterns(['action2*'])],
       action,
     });
+    new route53.ARecord(this, 'ARecord', {
+      target: route53.RecordTarget.fromAlias(new route53targets.LoadBalancerTarget(lb)),
+      zone: hostedZone,
+    });
 
     new CfnOutput(this, 'DNS', {
-      value: lb.loadBalancerDnsName,
+      value: props.domainName,
     });
   }
 }
 
+/**
+ * In order to test this you need to have a valid public hosted zone that you can use
+ * to request certificates for.
+ *
+*/
+const hostedZoneId = process.env.CDK_INTEG_HOSTED_ZONE_ID ?? process.env.HOSTED_ZONE_ID;
+if (!hostedZoneId) throw new Error('For this test you must provide your own HostedZoneId as an env var "HOSTED_ZONE_ID"');
+const hostedZoneName = process.env.CDK_INTEG_HOSTED_ZONE_NAME ?? process.env.HOSTED_ZONE_NAME;
+if (!hostedZoneName) throw new Error('For this test you must provide your own HostedZoneName as an env var "HOSTED_ZONE_NAME"');
+const domainName = process.env.CDK_INTEG_DOMAIN_NAME ?? process.env.DOMAIN_NAME;
+if (!domainName) throw new Error('For this test you must provide your own Domain Name as an env var "DOMAIN_NAME"');
+
 const app = new App();
-const testCase = new CognitoStack(app, 'integ-cognito');
-new integ.IntegTest(app, 'integ-test-cognito', {
+const testCase = new CognitoStack(app, 'integ-cognito', {
+  hostedZoneId,
+  hostedZoneName,
+  domainName,
+});
+const test = new integ.IntegTest(app, 'integ-test-cognito', {
   testCases: [testCase],
 });
-
+const testUser = new CognitoUser(testCase, 'User', {
+  userPool: testCase.userPool,
+  username: 'test-user@example.com',
+  password: 'TestUser@123',
+});
+// this function signs in to the website and returns text content of the authenticated page body
+const signinFunction = new lambda.Function(testCase, 'Signin', {
+  functionName: 'cdk-integ-alb-cognito-signin-handler',
+  code: lambda.Code.fromAsset('alb-cognito-signin-handler'),
+  handler: 'index.handler',
+  runtime: lambda.Runtime.NODEJS_18_X,
+  environment: {
+    TEST_USERNAME: testUser.username,
+    TEST_PASSWORD: testUser.password,
+    TEST_URL: `https://${domainName}`,
+  },
+  memorySize: 1024,
+  timeout: Duration.minutes(5),
+});
+const invoke = test.assertions.invokeFunction({
+  functionName: signinFunction.functionName,
+});
+invoke.expect(integ.ExpectedResult.objectLike({
+  Payload: '"Authenticated"',
+}));
```

Comment:
I added an assertion to sign in using puppeteer and verify authenticated page content.